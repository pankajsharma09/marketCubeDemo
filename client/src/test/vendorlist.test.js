import React from 'react';
import wait from 'waait'
import UserList from '../user/userList';
import { shallow, mount, configure } from 'enzyme';
import { MockedProvider } from "@apollo/react-testing";
import { gql } from 'apollo-boost';
import Adapter from 'enzyme-adapter-react-16';
import { AppProvider } from '@shopify/polaris';


configure({ adapter: new Adapter() });

const GET_QUERY = gql`
query UserLists{
    getUsers{
    id,
    firstName,
    lastName,
    email,
    brandName,
    status,
    createDate
    }
}
`;

const SUBSCRIPTION = gql`
    subscription  UserLists{
        newUserCreated {
            id,
            firstName,
            lastName,
            email,
            brandName,
            createDate,
            status
        }
    }
`

const mocks = [
    {
        request: {
            query: GET_QUERY,
        },
        result: {
            data: {
                getUsers: {
                    "id": "5da5a70f8a360b338519c1f2",
                    "firstName": null,
                    "lastName": null,
                    "email": "qwerty123@gmail.com",
                    "brandName": null,
                    "status": null,
                    "createDate": null
                  }
            },
        },
    },
    {
        request: {
            query: SUBSCRIPTION,
        },
        result: {
            data: {
                newUserCreated: {
                    "id": "5da5a70f8a360b338519c1fqq2",
                    "firstName": 'qwert',
                    "lastName": 'qwerty',
                    "email": "qwerty123@gmail.com",
                    "brandName": 'Something',
                    "status": 1,
                    "createDate": null
                  }
            },
        },
    },

];

describe('listing', () => {
    window.matchMedia = jest.fn().mockImplementation(query => {
        return {
          matches: false,
          media: query,
          onchange: null,
          addListener: jest.fn(),
          removeListener: jest.fn(),
        };
      });
    // const component = mount(
    //     <MockedProvider mocks={mocks} addTypename={false}>
    //         <AppProvider>
    //             <UserList />
    //         </AppProvider>
    //     </MockedProvider>
    // );
    const component = shallow(
        <MockedProvider mocks={mocks} addTypename={false}>
            <AppProvider>
                <UserList />
            </AppProvider>
        </MockedProvider>
    );
    
    it('render query', async () => {
        await wait(1000);
        console.log('qwert',component.dive().dive().debug());
        expect(component).toMatchSnapshot();
    });
})




// describe('Vendor Listing', () => {

//     // window.matchMedia = jest.fn().mockImplementation(query => {
//     //     return {
//     //       matches: f{alse},
//     //       media: query,
//     //       onchange: null,
//     //       addListener: jest.fn(),
//     //       removeListener: jest.fn(),
//     //     };
//     //   });
//     const component = render.create(
//             <MockedProvider mocks={mocks}>
//                     <UserList/>
//             </MockedProvider>
//     );
//     // it('render vendor listing', () => {
//     //     const tree = component.toJSON();

//     // });

// });
