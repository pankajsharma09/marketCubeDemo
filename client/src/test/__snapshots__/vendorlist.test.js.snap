// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listing render query 1`] = `ShallowWrapper {}`;
